src/main/riddl/Members/members.riddl

```diff
@@ -157,24 +211,126 @@ context MembersContext  is {
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example InactiveMemberToTerminatedMember is {
+                given "entity is transitioning from Inactive to Terminated and required fields are non-empty"
+                then "set entity Status to Status field from message"
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example InactiveMemberIllegalStateTransition is {
+                given "entity is transitioning from Draft to Draft, Inactive, or Suspended"
+                then error "Draft members may not transition to desired state."
+              }
+            }
+            on command UpdateMemberInfo {
+              then "set entity Info to non-Empty fields in UpdateInfo from message"
+              and "set entity lastModifiedBy to actingMember from message"
+              and "set entity lastModifiedOn to today()"
+              and "yield event MemberInfoUpdated"
+            }
+            on query GetMemberData {
+              then "yield Info for entity"
+            }
+          }
+        }
+
+        state Suspended is {
+          fields {
+            info: MembersContext.Info,
+            meta: MembersContext.MetaInfo
+          }
+
+          handler InactiveMemberHandler {
+            on command RegisterMember {then error "Member has already been registered."}
+            on command UpdateMemberStatus {
+              example SuspendedMemberToActiveMember is {
+                given "entity is transitioning from Suspended to Active and required fields are non-empty"
+                then "set entity Status to Status field from message"
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example SuspendedMemberToTerminatedMember is {
+                given "entity is transitioning from Suspended to Terminated and required fields are non-empty"
+                then "set entity Status to Status field from message"
+                and "set entity lastModifiedBy to actingMember from message"
+                and "set entity lastModifiedOn to today()"
+                and "yield event MemberStatusUpdated"
+              }
+              example SuspendedMemberIllegalStateTransition is {
+                given "entity is transitioning from Suspended to Draft, Inactive, or Suspended"
+                then error "Draft members may not transition to desired state."
+              }
+            }
+            on command UpdateMemberInfo {
```

Comment:
Do we also need to add an UpdateSuspendedReason command - or have a different aggregate (SuspendedMemberInfo?) to contain reason?

Reply:
Maybe Info should have an optional suspensionReason, and UpdateMemberStatus to have reason: String, where it is optional, but required for updating to the suspended state? These changes should allow us to have what you proposed

Reply:
The way James has modeled these entities is as state machines. Suspended is a state. The "reason" only needs to exist when the entity is in the Suspended state. So, I think UpdatedSuspensionReason is a reasonable command, but the Suspended state could also just accept another SuspendMember command and update the "reason" status as part of that.

Reply:
Alright I think that makes sense. I will add UpdateSuspensionReason and UpdatedSuspensionReason commands. As for storing the "reason" field, should this be added into the MetaInfo? I'm not too sure where this should be persisted.

Reply:
I would rather you just allowed the SuspendMember message to be handled when in the Suspended state.

Reply:
The "reason" field should be called "suspensionReason" and be part of the Suspended state's fields.

Reply:
The middle bullet is incorrect. It should be allowed in all states.  Whether you've got an Active or Draft state for member, they should be suspendable.

Reply:
As per our messages offline, here are the changes applied:

- Removed Status in members.riddl. 
- Removed memberStatus. 
- Removed UpdateMemberStatus. 
- Removed MemberStatusUpdated. 
- Add ActivateMember, MemberActivated, InactivateMember, MemberInactivated, SuspendMember, MemberSuspended, TerminateMember, MemberTerminated. 
- Add suspensionReason to the Suspended state.